x-pack/plugins/apm/public/hooks/use_fetcher.tsx

```diff
@@ -55,9 +56,12 @@ function getDetailsFromErrorResponse(
 
 const createAutoAbortedAPMClient = (
   signal: AbortSignal,
-  addInspectorRequest: <Data>(result: FetcherResult<Data>) => void
+  addInspectorRequest: <Data>(result: FetcherResult<Data>) => void,
+  setEndpoint: (endpoint: string) => void
 ): AutoAbortedAPMClient => {
   return ((endpoint, options) => {
+    setEndpoint(endpoint);
```

Comment:
are we actually using this? seems like we're specifying a name called `endpoint` in `useFetcherSpan`? Maybe I'm misunderstanding though

Reply:
Yes, here:
https://github.com/elastic/kibana/blob/3afa422584a6725346eda78e14211fba3edb3492/x-pack/plugins/apm/public/components/app/service_inventory/index.tsx#L253-L256

Reply:
`mainStatisticsFetch` and `comparisonFetch` are both `FETCH_RESULT` objects that contain the `endpoint` prop.

Reply:
ahh, got it. but why do the others need an endpoint override?

Reply:
Because they don't pass down the `FETCH_RESULT` - only status